@jiayuehua, I tested the usage of `fbthrift` locally and encountered such an error through `fbthrift:x64-windows-static`, please take a look:
````
1> CMake generation started for configuration: 'x64-static-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\ENTERPRISE\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 16 2019" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\CMakeProject42\out\install\x64-static-Debug" -DVCPKG_TARGET_TRIPLET:STRING="x64-windows-static" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/fb/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\CMakeProject42" 2>&1"
1> Working directory: C:\Users\test\source\repos\CMakeProject42\out\build\x64-static-Debug
1> [CMake] -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
1> [CMake] -- The C compiler identification is MSVC 19.29.30148.0
1> [CMake] -- The CXX compiler identification is MSVC 19.29.30148.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] -- Found ZLIB: optimized;F:/Feature-test/fb/vcpkg/installed/x64-windows-static/lib/zlib.lib;debug;F:/Feature-test/fb/vcpkg/installed/x64-windows-static/debug/lib/zlibd.lib (found version "1.2.13") 
1> [CMake] -- Found FBThrift: F:/Feature-test/fb/vcpkg/installed/x64-windows-static
1> [CMake] -- Configuring done
1> [CMake] CMake Error at F:/Feature-test/fb/vcpkg/scripts/buildsystems/vcpkg.cmake:591 (_add_executable):
1> [CMake]   Target "main" links to target "Folly::folly" but the target was not found.
1> [CMake]   Perhaps a find_package() call is missing for an IMPORTED target, or an
1> [CMake]   ALIAS target is missing?
1> [CMake] Call Stack (most recent call first):
1> [CMake]   CMakeProject42/CMakeLists.txt:14 (add_executable)
1> [CMake] CMake Error at F:/Feature-test/fb/vcpkg/scripts/buildsystems/vcpkg.cmake:591 (_add_executable):
1> [CMake]   Target "main" links to target "glog::glog" but the target was not found.
1> [CMake]   Perhaps a find_package() call is missing for an IMPORTED target, or an
1> [CMake]   ALIAS target is missing?
1> [CMake] Call Stack (most recent call first):
````

CMakeLists:
````
cmake_minimum_required (VERSION 3.8)
set(CMAKE_CXX_STANDARD 23)

project(test)

add_executable (main "CMakeProject42.cpp" "CMakeProject42.h")

find_package(FBThrift CONFIG REQUIRED)

target_link_libraries(main PRIVATE FBThrift::async FBThrift::thrift FBThrift::transport FBThrift::thriftcpp2)
````

--- a/ports/fbthrift/0002-fix-dependency.patch
+++ b/ports/fbthrift/0002-fix-dependency.patch
@@ -37,15 +37,20 @@ index beae7d7..0d03162 100644
    add_definitions("-DTHRIFT_HAVE_LIBSNAPPY=0")
    if(lib_only)
 diff --git a/thrift/cmake/FBThriftConfig.cmake.in b/thrift/cmake/FBThriftConfig.cmake.in
-index e1297ed..303c477 100644
+index e1297ed..c7d9d59 100644
 --- a/thrift/cmake/FBThriftConfig.cmake.in
 +++ b/thrift/cmake/FBThriftConfig.cmake.in
-@@ -28,7 +28,9 @@ else()
+@@ -28,7 +28,14 @@ else()
    set_and_check(FBTHRIFT_COMPILER "@PACKAGE_BIN_INSTALL_DIR@/thrift1")
  endif()
  
 -find_dependency(ZLIB REQUIRED)
++find_dependency(fizz CONFIG )
++find_dependency(fmt CONFIG )
++find_dependency(folly CONFIG )
 +find_dependency(gflags CONFIG)
++find_dependency(glog CONFIG )
++find_dependency(wangle CONFIG )
 +find_dependency(ZLIB)
 +find_dependency(zstd CONFIG)
  

--- a/ports/fbthrift/0002-fix-dependency.patch
+++ b/ports/fbthrift/0002-fix-dependency.patch
@@ -1,5 +1,5 @@
 diff --git a/CMakeLists.txt b/CMakeLists.txt
-index beae7d7..f962a1f 100644
+index beae7d7..0d03162 100644
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
 @@ -100,7 +100,8 @@ endif ()
@@ -17,7 +17,7 @@ index beae7d7..f962a1f 100644
    find_package(wangle CONFIG REQUIRED)
    find_package(ZLIB REQUIRED)
 -  find_package(Zstd REQUIRED)
-+  find_package(zstd REQUIRED)
++  find_package(zstd CONFIG REQUIRED)
 +  if(TARGET zstd::libzstd_shared)
 +    set(ZSTD_LIBRARIES zstd::libzstd_shared)
 +  elseif(TARGET zstd::libzstd_static)

--- a/ports/fbthrift/0002-fix-dependency.patch
+++ b/ports/fbthrift/0002-fix-dependency.patch
@@ -1,0 +1,53 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index beae7d7..f962a1f 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -100,7 +100,8 @@ endif ()
+ 
+ # Find required dependencies for thrift/lib
+ if(lib_only OR build_all)
+-  find_package(Gflags REQUIRED)
++  find_package(gflags CONFIG REQUIRED)
++  set(LIBGFLAGS_LIBRARY gflags::gflags)
+   find_package(glog CONFIG REQUIRED)
+   set (GLOG_LIBRARIES glog::glog)
+   find_package(folly CONFIG REQUIRED)
+@@ -108,16 +109,17 @@ if(lib_only OR build_all)
+   find_package(fmt CONFIG REQUIRED)
+   find_package(wangle CONFIG REQUIRED)
+   find_package(ZLIB REQUIRED)
+-  find_package(Zstd REQUIRED)
++  find_package(zstd REQUIRED)
++  if(TARGET zstd::libzstd_shared)
++    set(ZSTD_LIBRARIES zstd::libzstd_shared)
++  elseif(TARGET zstd::libzstd_static)
++    set(ZSTD_LIBRARIES zstd::libzstd_static)
++  endif()
++
+   # https://cmake.org/cmake/help/v3.9/module/FindThreads.html
+   set(THREADS_PREFER_PTHREAD_FLAG ON)
+   find_package(Threads)
+   include_directories(
+-    ${LIBGFLAGS_INCLUDE_DIR}
+-    ${GLOG_INCLUDE_DIRS}
+-    ${OPENSSL_INCLUDE_DIR}
+-    ${ZSTD_INCLUDE_DIRS}
+-    ${Boost_INCLUDE_DIRS}
+   )
+   add_definitions("-DTHRIFT_HAVE_LIBSNAPPY=0")
+   if(lib_only)
+diff --git a/thrift/cmake/FBThriftConfig.cmake.in b/thrift/cmake/FBThriftConfig.cmake.in
+index e1297ed..303c477 100644
+--- a/thrift/cmake/FBThriftConfig.cmake.in
++++ b/thrift/cmake/FBThriftConfig.cmake.in
+@@ -28,7 +28,9 @@ else()
+   set_and_check(FBTHRIFT_COMPILER "@PACKAGE_BIN_INSTALL_DIR@/thrift1")
+ endif()
+ 
+-find_dependency(ZLIB REQUIRED)
++find_dependency(gflags CONFIG)
++find_dependency(ZLIB)
++find_dependency(zstd CONFIG)
+ 
+ if (NOT TARGET FBThrift::thriftcpp2)
+   include("${FBTHRIFT_CMAKE_DIR}/FBThriftTargets.cmake")

--- a/ports/fbthrift/portfile.cmake
+++ b/ports/fbthrift/portfile.cmake
@@ -10,8 +10,18 @@ vcpkg_from_github(
     PATCHES 
         0001-fix-compatibility-with-boost-1.79.0.patch
         fix-glog.patch
+        0002-fix-dependency.patch
 )
 
+file(REMOVE "${SOURCE_PATH}/thrift/cmake/FindGMock.cmake")
+file(REMOVE "${SOURCE_PATH}/thrift/cmake/FindOpenSSL.cmake")
+file(REMOVE "${SOURCE_PATH}/thrift/cmake/FindZstd.cmake")
+file(REMOVE "${SOURCE_PATH}/build/fbcode_builder/CMake/FindGflags.cmake")
+file(REMOVE "${SOURCE_PATH}/build/fbcode_builder/CMake/FindGlog.cmake")
+file(REMOVE "${SOURCE_PATH}/build/fbcode_builder/CMake/FindGMock.cmake")
+file(REMOVE "${SOURCE_PATH}/build/fbcode_builder/CMake/FindLibEvent.cmake")
+file(REMOVE "${SOURCE_PATH}/build/fbcode_builder/CMake/FindSodium.cmake")
+file(REMOVE "${SOURCE_PATH}/build/fbcode_builder/CMake/FindZstd.cmake")
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS

--- a/ports/fbthrift/vcpkg.json
+++ b/ports/fbthrift/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "fbthrift",
   "version-string": "2022.03.21.00",
-  "port-version": 2,
+  "port-version": 3,
   "description": "Facebook's branch of Apache Thrift, including a new C++ server.",
   "homepage": "https://github.com/facebook/fbthrift",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2438,7 +2438,7 @@
     },
     "fbthrift": {
       "baseline": "2022.03.21.00",
-      "port-version": 2
+      "port-version": 3
     },
     "fcl": {
       "baseline": "0.7.0",

--- a/versions/f-/fbthrift.json
+++ b/versions/f-/fbthrift.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "b574cd22e85718b498dd1103dd3c915cb9090791",
+      "version-string": "2022.03.21.00",
+      "port-version": 3
+    },
+    {
       "git-tree": "60135cc9f7a4bf698b67d275e1ad66a542d9acb4",
       "version-string": "2022.03.21.00",
       "port-version": 2

--- a/versions/f-/fbthrift.json
+++ b/versions/f-/fbthrift.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "fd4683b3d6950cb569e21c57716cb1b9d66e592c",
+      "git-tree": "74805d5c9147b3255703cdce511d9082cdfee016",
       "version-string": "2022.03.21.00",
       "port-version": 3
     },

--- a/versions/f-/fbthrift.json
+++ b/versions/f-/fbthrift.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "b574cd22e85718b498dd1103dd3c915cb9090791",
+      "git-tree": "fd4683b3d6950cb569e21c57716cb1b9d66e592c",
       "version-string": "2022.03.21.00",
       "port-version": 3
     },

